packages/calcite-components/src/components/date-picker/date-picker.tsx

```diff
@@ -224,14 +215,37 @@ export class DatePicker extends LitElement {
       this.valueAsDateWatcher(this.valueAsDate);
     }
 
-    if (changes.has("min")) {
+    let minSource: Extract<keyof DatePicker, "min" | "minAsDate">;
+    let maxSource: Extract<keyof DatePicker, "max" | "maxAsDate">;
+
+    if (changes.has("min") && !changes.has("minAsDate")) {
+      minSource = "min";
+    } else if (changes.has("minAsDate") && !changes.has("min")) {
+      minSource = "minAsDate";
+    }
+
+    if (changes.has("max") && !changes.has("maxAsDate")) {
+      maxSource = "max";
+    } else if (changes.has("maxAsDate") && !changes.has("max")) {
+      maxSource = "maxAsDate";
+    }
+
+    if (minSource === "min") {
       this.onMinChanged(this.min);
     }
 
-    if (changes.has("max")) {
+    if (maxSource === "max") {
       this.onMaxChanged(this.max);
     }
 
+    if (minSource === "minAsDate") {
+      this.onMinChanged(dateToISO(this.minAsDate));
+    }
+
+    if (maxSource === "maxAsDate") {
+      this.onMaxChanged(dateToISO(this.maxAsDate));
```

Comment:
Can we avoid calling this.onMinChanged & this.onMaxChanged here and call this.setActiveStartAndEndDates method.

```
if (this.range && (minSource === "minAsDate" || maxSource === "maxAsDate" )) {
this.setActiveStartAndEndDates();
}